### PR TITLE
Fixes #1783 catalog review

### LINKED ISSUE #1783: Review Catalog UI
A mockup for a minimal review [here](https://drive.google.com/file/d/0B9nuKyTvUeXcb0hpaUVnd0R0c0E/view)
In milestone 2017.06.00 we will consider a complete review.

### PR DESCRIPTION
**Pull request by MV88**
see #1783

### CODE REVIEW

File: web/client/actions/catalog.js [hunk 1]
```diff
@@ -22,6 +22,14 @@ const RESET_CATALOG = 'RESET_CATALOG';
 const RECORD_LIST_LOAD_ERROR = 'RECORD_LIST_LOAD_ERROR';
 const CHANGE_CATALOG_FORMAT = 'CHANGE_CATALOG_FORMAT';
 const ADD_LAYER_ERROR = 'ADD_LAYER_ERROR';
+const CHANGE_SELECTED_SERVICE = 'CATALOG:CHANGE_SELECTED_SERVICE';
```
Comment (mbarto) [inline web/client/actions/catalog.js]: maybe we should prefix also the existing actions, if we decide to go that way (we decided that?)

File: web/client/actions/catalog.js [hunk 2]
```diff
@@ -65,6 +121,9 @@ function getRecords(format, url, startPosition = 1, maxRecords, filter, options)
             }
         }).catch((e) => {
             dispatch(recordsLoadError(e));
+            /*eslint-disable*/
+            console.log(e)
```
Comment (mbarto) [inline web/client/actions/catalog.js]: Please, remove console logging

File: web/client/actions/catalog.js [hunk 3]
```diff
@@ -84,6 +143,9 @@ function textSearch(format, url, startPosition, maxRecords, text, options) {
...
```
Comment (mbarto) [inline web/client/actions/catalog.js]: Please, remove console logging

File: web/client/components/catalog/Catalog.jsx [hunk 4]
```diff
@@ -1,77 +1,117 @@
...
+        onFocusServicesList: () => {},
+        onLayerAdd: () => {},
+        onReset: () => {},
+        onSearch: () => {},
+        onZoomToExtent: () => {},
+        openCatalogServiceList: false,
+        pageSize: 4,
+        records: [],
+        services: {},
+        serviceComboboxMessages: {
```
Comment (mbarto) [inline web/client/components/catalog/Catalog.jsx]: Do we really need to make these configurable?

File: web/client/components/catalog/Catalog.jsx [hunk 5]
```diff
@@ -139,7 +179,7 @@ class Catalog extends React.Component {
         let pageSize = this.props.pageSize;
         let page = Math.floor( start / pageSize);
         let pageN = Math.ceil(total / pageSize);
-        return (<div><Pagination
+        return (<div style={{textAlign: "center"}}><Pagination
```
Comment (mbarto) [inline web/client/components/catalog/Catalog.jsx]: Can we use a class and CSS so that this is overridable?

File: web/client/epics/catalog.js [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+                            });
+                        }
+                    }
+                    return {notification, addNewService};
+                }))
+                .switchMap((actions) => {
+                    return actions.addNewService !== null ? Rx.Observable.of(actions.notification, actions.addNewService) : Rx.Observable.of(actions.notification);
+                }).catch((e) => {
+                    /*eslint-disable*/
+                    console.log("error fetching getCapabilites of " + API[newService.type].parseUrl(newService.url) + " \nthe error is:\n");
```
Comment (mbarto) [inline web/client/epics/catalog.js]: Please remove console logging

File: web/client/product/appConfig.js [hunk 7]
```diff
@@ -60,6 +60,31 @@ module.exports = {
                     enabled: false,
                     seeMore: false
                 }
+            },
+            catalog: {
+                "selectedService": "",
+                "services": {
```
Comment (mbarto) [inline web/client/product/appConfig.js]: this should go in localConfig

File: web/client/selectors/catalog.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
+const {get} = require('lodash');
+
+module.exports = {
+    resultSelector: (state) => get(state, `catalog.result`),
```
Comment (mbarto) [inline web/client/selectors/catalog.js]: why backticks?

Comment (MV88) [reply web/client/selectors/catalog.js]: They are only needed here:
selectedServiceTypeSelector: (state) => get(state, `catalog.services.${get(state, "catalog.selectedService")}.type`, 'csw')

i'll replace them with use " in the other cases

File: web/client/localConfig.json [hunk 9]
```diff
@@ -30,6 +30,35 @@
...
       {"name": "featuregridmode", "path": "featuregrid.mode"}],
+    "initialState": {
+      "defaultState": {
+        "catalog":{
+          "selectedService": "",
+          "services": {
+              "Basic CSW Service": {
+                  "url": "https://demo.geo-solutions.it/geoserver/csw",
+                  "type": "csw",
+                  "title": "Basic CSW Service"
```
Comment (offtherailz) [inline web/client/localConfig.json]: Call them "Demo CSW Service" etc...

**Comment by MV88** [comment]
closing this, opening a new PR with conflicts resolved.

**Pull request closed without merging**